.nomad/apiary.nomad

```diff
@@ -0,0 +1,337 @@
+        data = <<EOF
+DOCKER_IMAGE_DIGEST="${split("@", var.image)[1]}"
+PERSIST_RESUMES="${var.persist_resumes}"
+EOF
+
+        destination = "/secrets/.docker_image_digest"
+        env = true
+      }
+
+      template {
+        data = trimspace(file("conf/.my.cnf"))
+
+        destination = "/secrets/.my.cnf"
+
+        change_mode = "noop"
+      }
+    }
+
+    task "web" {
+      driver = "docker"
+
+      config {
+        image = var.image
+
+        force_pull = true
+
+        network_mode = "host"
+
+        mount {
+          type   = "bind"
+          source = "local/fpm/"
+          target = "/etc/php/7.4/fpm/pool.d/"
+        }
+
+        mount {
+          type   = "bind"
+          source = "local/"
+          target = "/app/storage/logs/"
+        }
+
+        mount {
+          type = "tmpfs"
+          target = "/run/php"
+          readonly = false
+          tmpfs_options {
+            size = 16000
+          }
+        }
+
+        entrypoint = [
+          "/bin/bash",
+          "-xeuo",
+          "pipefail",
+          "-c",
+          trimspace(file("scripts/web.sh"))
+        ]
+      }
+
+      resources {
+        cpu = 500
+        memory = 512
+        memory_max = 2048
+      }
+
+      volume_mount {
+        volume = "run"
+        destination = "/var/opt/nomad/run/"
+      }
+
+      dynamic "volume_mount" {
+        for_each = var.persist_resumes ? ["resumes"] : []
+
+        content {
+          volume = "resumes"
+          destination = "/app/storage/app/resumes/"
+        }
+      }
+
+      template {
+        data = trimspace(file("conf/www.conf"))
+
+        destination = "local/fpm/www.conf"
+      }
+
+      template {
+        data = trimspace(file("conf/.env.tpl"))
+
+        destination = "/secrets/.env"
+        env = true
+      }
+
+      template {
+        data = "DOCKER_IMAGE_DIGEST=\"${split("@", var.image)[1]}\""
+
+        destination = "/secrets/.docker_image_digest"
+        env = true
+      }
+
+      template {
+        data = trimspace(file("conf/.my.cnf"))
+
+        destination = "/secrets/.my.cnf"
+
+        change_mode = "noop"
+      }
+
+      service {
+        name = "${NOMAD_JOB_NAME}"
+
+        check {
+          name = "GET /ping"
+
+          type = "script"
+
+          command = "/bin/bash"
+
+          args = [
+            "-euxo",
+            "pipefail",
+            "-c",
+            trimspace(file("scripts/healthcheck.sh"))
+          ]
+
+          interval = "5s"
+          timeout = "5s"
+        }
+
+        check_restart {
+          limit = 5
+          grace = "20s"
+        }
+
+        meta {
+          nginx-config = trimspace(trimsuffix(trimspace(regex_replace(regex_replace(regex_replace(regex_replace(regex_replace(regex_replace(regex_replace(regex_replace(trimspace(file("conf/nginx.conf")),"server\\s{\\s",""),"server_name\\s\\S+;",""),"root\\s\\S+;",""),"listen\\s.+;",""),"#.+\\n",""),";\\s+",";"),"{\\s+","{"),"\\s+"," ")),"}"))
```

Comment:
What is this function vomit?

Reply:
This is essentially a lossless compression algorithm for Nginx configuration.

Meta tag values are limited to 512 characters, so this strips out whitespace, comments, and some lines that are populated by the nginx job itself (`server`, `server_name`, `listen`, `root`).

The point of all this is so Nginx configuration can be version-controlled and promoted between environments, like any other artifact.